### PR TITLE
Added confirmation of existence of email when updating user info

### LINKED ISSUE #250: Bug: Unable to log in when email and password pair overlap
Environment
------------

- [crowi] 1.6.2

How to reproduce? (再現手順)
---------------------------

1. Register user1
    * email: `user1@example.com`
    * password: `password1`
1. Register user2
    * email: `user2@example.com`
    * password: `password2`
1. Login with `user1@example.com` (logged in as user1) and modify credentials
    1. Go to `/me` and change e-mail to `user2@example.com`
    1. Go to `/me/password` and change password to `password2`
1. Logout
1. Login with `user2@example.com` and `password2` pair

What happens? (症状)
---------------------

- user1 can log in
    - `userSchema.statics.findUserByEmailAndPassword` picks the old one
- user2 is unable to log in until user1 modify e-mail or password

What is the expected result? (期待される動作)
-------------------------------------------

- user1 cannot modify e-mail to `user2@example.com`
    - should be checked whether the e-mail is reserved

Note
----

- Mongoose can attach unique constraint
    - http://mongoosejs.com/docs/2.7.x/docs/indexes.html
- If `User.email` becomes unique, `statusDeleted` method will be affected
https://github.com/crowi/crowi/blob/56e8fe76aab2bde75590c4cf08b1503812932a62/lib/models/user.js#L242

### PR DESCRIPTION
**Pull request by suzuki**
## About

- fix #250

### CODE REVIEW

File: lib/routes/me.js [hunk 1]
```diff
@@ -89,20 +89,29 @@ module.exports = function(crowi, app) {
...
+          req.form.errors.push('It can not be changed to that mail address');
           return res.render('me/index', {});
         }
 
-        req.i18n.changeLanguage(lang);
-        req.flash('successMessage', req.t('Updated'));
-        return res.redirect('/me');
+        userData.update(name, email, lang, (err, userData) => {
+          if (err) {
+            for (var e in err.errors) {
```
Comment (abouthiroppy) [inline lib/routes/me.js]: You should write `Object.keys(err.errors).forEach(() => ...)` and delete L:102.

File: lib/routes/me.js [hunk 2]
```diff
@@ -89,20 +89,29 @@ module.exports = function(crowi, app) {
...
 
-      userData.update(name, email, lang, function(err, userData) {
-        if (err) {
-          for (var e in err.errors) {
-            if (err.errors.hasOwnProperty(e)) {
-              req.form.errors.push(err.errors[e].message);
-            }
-          }
+      User.findOne({email: email}, (err, existingUserData) => {
+        if (existingUserData !== null) {
```
Comment (abouthiroppy) [inline lib/routes/me.js]: `if (existingUserData)` is better;)

Comment (suzuki) [reply lib/routes/me.js]: 👍

File: lib/routes/me.js (hunk 1 above)
Comment (suzuki) [reply lib/routes/me.js]: Thanks! I'll fix.

File: lib/routes/me.js [hunk 3]
```diff
@@ -89,20 +89,27 @@ module.exports = function(crowi, app) {
...
+          req.form.errors.push('It can not be changed to that mail address');
           return res.render('me/index', {});
         }
 
-        req.i18n.changeLanguage(lang);
-        req.flash('successMessage', req.t('Updated'));
-        return res.redirect('/me');
+        userData.update(name, email, lang, (err, userData) => {
+          if (err) {
+            Object.keys(err.errors).forEachh((e) => {
```
Comment (abouthiroppy) [inline lib/routes/me.js]: typo

```diff
- Object.keys(err.errors).forEachh((e) => {
+ Object.keys(err.errors).forEach((e) => {
```

Comment (suzuki) [reply lib/routes/me.js]: 😱

**Comment by sotarok** [comment]
Thank you for fixing the problem. I'll check it soon.

**Pull request merged**